sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/_tests_/student-loan-balances-queue.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,114 @@
+import * as path from "path";
+import { StudentLoanBalancesScheduler } from "../student-loan-balances-queue.scheduler";
+import { StudentLoanBalances } from "@sims/sims-db";
+
+// Student loan balances received file mocks.
+const STUDENT_LOAN_BALANCES_FILENAME = "PEDU.PBC.PT.MBAL.20240302.001";
+
+describe(describeProcessorRootTest(QueueNames.StudentLoanBalances), () => {
+  let app: INestApplication;
+  let processor: StudentLoanBalancesScheduler;
+  let db: E2EDataSources;
+  let sftpClientMock: DeepMocked<Client>;
+  let studentLoanBalancesDownloadFolder: string;
+
+  beforeAll(async () => {
+    studentLoanBalancesDownloadFolder = path.join(
+      __dirname,
+      "student-loan-balances-files",
+    );
+    // Set the ESDC response folder to the files mocks folders.
+    process.env.ESDC_RESPONSE_FOLDER = studentLoanBalancesDownloadFolder;
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(StudentLoanBalancesScheduler);
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it("Should add monthly loan balance record for the student when the student matches SIN, DOB, and last name.", async () => {
+    // Arrange
+    // Create student if it doesn't exist.
+    let student = await db.student.findOne({
+      where: {
+        birthDate: getISODateOnlyString(new Date("1998-03-24")),
+        user: { lastName: "LASTNAME" },
+        sinValidation: { sin: "900041310" },
+      },
+    });
+    if (!student) {
+      student = await saveFakeStudent(db.dataSource);
+      // Update the student to ensure that the student receiving loan balance is the same student as the one created above.
+      student.birthDate = getISODateOnlyString(new Date("1998-03-24"));
+      student.user.lastName = "LASTNAME";
+      student.sinValidation.sin = "900041310";
+      await db.student.save(student);
+    }
+    db.studentLoanBalances.delete({ studentId: student.id });
+    // Queued job.
+    const { job } = mockBullJob<void>();
+    mockDownloadFiles(sftpClientMock, [STUDENT_LOAN_BALANCES_FILENAME]);
+    // Act
+    const result = await processor.processStudentLoanBalancesFiles(job);
+    // Assert
+    expect(result.length).toBe(1);
+    expect(result).toContain("Process finalized with success.");
+    // Expect the file was deleted from SFTP.
+    expect(sftpClientMock.delete).toHaveBeenCalled();
+    const studentLoanBalances: StudentLoanBalances[] =
+      await db.studentLoanBalances.find({
+        where: {
+          studentId: student.id,
+        },
+      });
+    // Expect student loan balance contains the student record.
+    expect(studentLoanBalances.length).toBe(1);
+    expect(studentLoanBalances[0].balanceDate).toBe("2023-12-31");
+    expect(studentLoanBalances[0].cslBalance).toBe(148154.0);
+  });
```

Comment:
It can probably follow other E2E tests and execute the validation as below.
```ts
expect(studentLoanBalances).toStrictEqual([
{
   balanceDate: "2023-12-31",
   cslBalance: 148154.0,
  },
]);
```